### PR TITLE
Add dapr upgrade command

### PR DESCRIPTION
**Pull request by yaron2**
This PR adds a `dapr upgrade -k` command that greatly simplifies the Dapr upgrade experience.

This PR handles CRD updates and certificate exports for zero downtime upgrades.

Closes https://github.com/dapr/cli/issues/560

### CODE REVIEW

File: pkg/kubernetes/upgrade.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	"configuration",
+	"subscription",
+}
+
+type UpgradeConfig struct {
+	RuntimeVersion string
+}
+
+func Upgrade(conf UpgradeConfig) error {
+	helmConf, err := helmConfig("dapr-system")
```
Comment (mukundansundar) [inline pkg/kubernetes/upgrade.go:30]: Add `--namespace` flag as it can be a different namespace.

File: pkg/kubernetes/upgrade.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+	upgradeClient := helm.NewUpgrade(helmConf)
+	upgradeClient.ResetValues = true
+	upgradeClient.Namespace = status[0].Namespace
+	upgradeClient.CleanupOnFail = true
+	upgradeClient.Wait = true
+
+	print.InfoStatusEvent(os.Stdout, "Starting upgrade...")
+
+	mtls, err := IsMTLSEnabled()
```
Comment (mukundansundar) [inline pkg/kubernetes/upgrade.go:64]: should using old certs during upgrade be a flag? `--mtls` Only export certs during upgrade if that flag is provided.

**Review by mukundansundar — DISMISSED** [review]

**Comment by mukundansundar** [comment]
Please add unit tests for this change.

File: pkg/kubernetes/upgrade.go (hunk 2 above)
Comment (yaron2) [reply pkg/kubernetes/upgrade.go:64]: Its our recommended best practice to carry the certs over. In this PR, if mTLS is used, they are carried over.

File: pkg/kubernetes/upgrade.go (hunk 1 above)
Comment (yaron2) [reply pkg/kubernetes/upgrade.go]: updated

**Comment by yaron2** [comment]
@mukundansundar please review again.

File: pkg/kubernetes/upgrade.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+	status, err := sc.Status()
+	if err != nil {
+		return err
+	}
+
+	if len(status) == 0 {
+		return errors.New("dapr is not installed in your cluster")
+	}
+
+	print.InfoStatusEvent(os.Stdout, "Dapr control plane version %s detected", status[0].Version)
```
Comment (mukundansundar) [inline pkg/kubernetes/upgrade.go:44]: probably print the Namespace also here? 
Can there be more than one dapr installation in a cluster?

File: pkg/kubernetes/upgrade.go [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+			return err
+		}
+	}
+
+	err = applyCRDs(fmt.Sprintf("v%s", conf.RuntimeVersion))
+	if err != nil {
+		return err
+	}
+
+	if _, err = upgradeClient.Run("dapr", daprChart, vals); err != nil {
```
Comment (mukundansundar) [inline pkg/kubernetes/upgrade.go:93]: make a note somewhere that `dapr` is the release name that it looks to upgrade in helm. 
If it is installed under a different release name in helm and then upgrade is used it won't work. 
Can a `--release-name` flag be provided?

File: pkg/kubernetes/upgrade.go [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+		secret, sErr := getTrustChainSecret()
+		if sErr != nil {
+			return sErr
+		}
+
+		ca := secret.Data["ca.crt"]
+		issuerCert := secret.Data["issuer.crt"]
+		issuerKey := secret.Data["issuer.key"]
+
+		ha := highAvailabilityEnabled(status)
```
Comment (mukundansundar) [inline pkg/kubernetes/upgrade.go:81]: Can't ha be independent of mttls? So shouldn't ha values in the chart be set independently?

**Review by mukundansundar — COMMENTED** [review]
@yaron2 added some comments

File: pkg/kubernetes/upgrade.go (hunk 5 above)
Comment (yaron2) [reply pkg/kubernetes/upgrade.go]: Changed

File: pkg/kubernetes/upgrade.go (hunk 4 above)
Comment (yaron2) [reply pkg/kubernetes/upgrade.go:96]: I don't want to put Helm specific constructs in the command, also I don't think there's a need here: Helm users will know to list the installation with `helm list`, so specifying the release seems redundant here. If a user installed Dapr through Helm with a non-default release name, they should upgrade using Helm.

File: pkg/kubernetes/upgrade.go (hunk 3 above)
Comment (yaron2) [reply pkg/kubernetes/upgrade.go]: There can't be more than one Dapr installation in a cluster, but I added the namespace for good measure.

File: pkg/kubernetes/upgrade.go (hunk 4 above)
Comment (mukundansundar) [reply pkg/kubernetes/upgrade.go:96]: Can you make a note regarding that in README? 
Also there needs to be addition of upgrade command in the rc3 docs.

File: README.md [hunk 6]
```diff
@@ -211,6 +211,18 @@ To remove Dapr from your Kubernetes cluster, use the `uninstall` command with `-
 $ dapr uninstall -k
 ```
 
+### Upgrade Dapr on Kubernetes
+
+To perform a zero downtime upgrade of the Dapr control plane:
```
Comment (mukundansundar) [inline README.md:216]: Make a note of the release name being `dapr` and if installed using helm, should use helm to upgrade.

Comment (yaron2) [reply README.md:216]: Added a note for the installation, where it belongs.

File: pkg/kubernetes/upgrade.go (hunk 4 above)
Comment (yaron2) [reply pkg/kubernetes/upgrade.go]: This is now changed, in last commit I changed the behavior and now the upgrade command will work with any Helm installation of Dapr.

**Review by mukundansundar — APPROVED** [review]

**Pull request merged**